### PR TITLE
Split stop transaction

### PR DESCRIPTION
**Pull request by rmangin-fr**
tested :)

### CODE REVIEW

**Comment by LucasBrazi06** [comment]
Apparently the button to connect the issue is back, do not use Link Issues anymore:

I just did it :)

File: src/server/rest/v1/validator/ChargingStationValidator.ts [hunk 1]
```diff
@@ -89,6 +91,12 @@ export default class ChargingStationValidator extends SchemaValidator {
     return data;
   }
 
+  public validateChargingStationActionStopTransactionReq(data: HttpChargingStationCommandRequest): HttpChargingStationCommandRequest {
```
Comment (LucasBrazi06) [inline src/server/rest/v1/validator/ChargingStationValidator.ts:94]: I haven't noticed but I would expect a typed response here like HttpChargingStationStartTransactionRequest with the args prop typed.

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/types/requests/HttpChargingStationRequest.ts [hunk 2]
```diff
@@ -111,6 +111,12 @@ export interface HttpChargingStationCommandRequest {
   args?: any;
 }
 
+export interface HttpChargingStationCommandStopTransactionRequest {
+  chargingStationID: string,
+  args?: {
```
Comment (LucasBrazi06) [inline src/types/requests/HttpChargingStationRequest.ts:116]: 'arg' is not optional, remove the '?' :-)

File: src/types/requests/HttpChargingStationRequest.ts [hunk 3]
```diff
@@ -111,6 +111,12 @@ export interface HttpChargingStationCommandRequest {
...
+    transactionID: string
+  }
+}
 export interface HttpIsAuthorizedRequest {
```
Comment (LucasBrazi06) [inline src/types/requests/HttpChargingStationRequest.ts:120]: Add BL

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
Check my comments and do also the same for Start Transaction ;-)

File: src/server/rest/v1/validator/ChargingStationValidator.ts (hunk 1 above)
Comment (rmangin-fr) [reply src/server/rest/v1/validator/ChargingStationValidator.ts]: done

File: src/server/rest/v1/validator/ChargingStationValidator.ts [hunk 4]
```diff
@@ -83,12 +85,18 @@ export default class ChargingStationValidator extends SchemaValidator {
...
   }
 
-  public validateChargingStationActionStartTransactionReq(data: HttpChargingStationCommandRequest): HttpChargingStationCommandRequest {
+  public validateChargingStationActionStartTransactionReq(data: HttpChargingStationCommandStartTransactionRequest): HttpChargingStationCommandStartTransactionRequest {
     // Validate schema for StartTransaction
     this.validate(this.chargingStationActionStartTransaction, data);
     return data;
   }
 
+  public validateChargingStationActionStopTransactionReq(data: HttpChargingStationCommandStopTransactionRequest): HttpChargingStationCommandStopTransactionRequest {
```
Comment (LucasBrazi06) [inline src/server/rest/v1/validator/ChargingStationValidator.ts:94]: Without Command as I wrote ;-)
HttpChargingStationCommandStopTransactionRequest -> HttpChargingStationStopTransactionRequest

File: src/types/requests/HttpChargingStationRequest.ts [hunk 5]
```diff
@@ -111,6 +111,20 @@ export interface HttpChargingStationCommandRequest {
...
 
+export interface HttpChargingStationCommandStartTransactionRequest {
+  chargingStationID: string,
+  args: {
+    tagID?: string,
+    visualTagID?: string,
+    connectorId: string
+  }
+}
+export interface HttpChargingStationCommandStopTransactionRequest {
```
Comment (LucasBrazi06) [inline src/types/requests/HttpChargingStationRequest.ts:122]: Add BL

File: src/types/requests/HttpChargingStationRequest.ts (hunk 3 above)
Comment (LucasBrazi06) [reply src/types/requests/HttpChargingStationRequest.ts:128]: Still Add BL

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
Check my comments

**Review by LucasBrazi06 — APPROVED** [review]
GUT

**Pull request merged**